bip-0182.md

```diff
@@ -0,0 +1,369 @@
+perform the `BIP-0030` check. In theory, those blocks can't be reorged, because
+of checkpoints, that goes back to block height 295,000 with the block hash
+`00000000000000004d9b4ef50f0f9d686fd69db2e03af35a100370c64632a983`. Any chain that
+doesn't include this block at height 295,000 isn't valid as removing this check
+would be a hard-fork. We note, however, that after version `0.30`, Bitcoin Core
+will remove the checkpoints[^1], as they are not needed anymore to prevent attacks
+against nodes during Initial Block Download. This is effectively a hard-fork,
+that will probably never actually happen, however.
+
+Block 1,983,702 is the first block that Utreexo nodes would be in danger of a
+consensus failure due to the inability to perform the BIP-0030 checks if someone were
+to reuse coinbase transaction from block 164,384. However, this block will happen in roughly
+21 years from now, and some mitigations have been proposed [^2].
+
+### Historical BIP-0030 violations
+
+There were two UTXOs that were overwritten by repeated transactions:
+`e3bf3d07d4b0375638d5f1db5255fe07ba2c4cb067cd81b84ee974b6585fb468:0` at block height 91,722
+`d5d27987d2a3dfc724e359870c6644b40e497bdc0589a033220fe15429d88599:0` at block height 91,812
+
+Since the leaf hashes that are committed to the Utreexo accumulator commit to
+the block hash as well, all the leaf hashes are unique and the two historical
+violations do not happen with how the UTXO set is represented with the Utreexo
+accumulator. To be consensus compatible with clients that retain only the second
+occurrences of these outputs, the leaves representing the corresponding first UTXOs in the Utreexo accumulator
+are hardcoded as unspendable.
+
+These two leaf hashes encoded in hex string are:
+
+ 1. `84b3af0783b410b4564c5d1f361868559f7cf77cfc65ce2be951210357022fe3`
+ 2. `bc6b4bf7cebbd33a18d6b0fe1f8ecc7aa5403083c39ee343b985d51fd0295ad8`
+
+(1) represents the UTXO created at block height 91,722 and (2) represents the
+UTXO created at block height 91,812.
+
+## Rationale
+
+**Why use the Utreexo accumulator to keep track of UTXOs instead of a key-value database like leveldb?**
+
+There's two main advantages to using the Utreexo accumulator instead of a key-value database like leveldb:
+
+ 1. Puts a cap on the UTXO set growth.
+ 3. Performance gains with the elimination of random reads/writes.
+
+### Puts a cap on the UTXO set growth
+
+The UTXO set is the collection of unspent transaction outputs and is used to verify blocks.
+As the amount of Bitcoin users grow, the UTXO set grows with it.
+
+The UTXO set is currently around 10GB in 2025 and with pruning that's all it takes to maintain a full node.
+However, as the UTXO set grows, the disk storage requirement will grow along with it and increase the barrier to running a full node.
+
+Currently, the UTXO set size is $O(log(N))$ where $N$ is the number of UTXOs.
+By utilizing the Utreexo accumulator, we're able to cap the UTXO set growth at $O(log_2(N))$.
```

Comment:
Given that you don’t store the UTXO set, but an accumulator that commits to the UTXO set, perhaps these two sentences should be amended?